lib/SILGen/SILGenProlog.cpp

```diff
@@ -490,11 +490,19 @@ uint16_t SILGenFunction::emitProlog(ParameterList *paramList,
   // Add the SILArguments and use them to initialize the local argument
   // values.
   if (paramList)
-    for (auto *param : *paramList)
+    for (auto *param : *paramList) {
+      if (param->getType()->isStructurallyUninhabited()) {
+        SILLocation unreachableLoc(param);
+        unreachableLoc.markAsPrologue();
+        B.createUnreachable(unreachableLoc);
+      }
+      
```

Comment:
I think you’re emitting the unreachable too early in the functiok emission. I should’ve clarified the specific function I was mentioning. I think you should look at `void SILGenFunction::emitProlog` rather than the one that returns the arg count. In this function, right after you see the `emitProlog` call, is where I think you can check if any parameters have an uninhabited type and emit unreachable.

Reply:
Thanks for the suggestions, it no longer segfaults now!